dts/bindings/i2c/cdns,i2c-r1p14.yaml

```
@@ -0,0 +1,43 @@
+# Copyright (c) 2025 Advanced Micro Devices, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Cadence I2C controller version 1.4.
+
+  Example I2C device tree node:
+    i2c@e0004000 {
+        compatible = "cdns,i2c-r1p14";
+        clocks = <&clkc 38>;
+        interrupts = <GIC_SPI 25 IRQ_TYPE_LEVEL_HIGH>;
+        reg = <0xe0004000 0x1000>;
+        clock-frequency = <400000>;
+        #address-cells = <1>;
+        #size-cells = <0>;
+        fifo-depth = <8>;
+    };
+
+compatible: "cdns,i2c-r1p14"
```

I'd rather not see this with the IP revision number at the end of this. I have also seen the 1.24 version of this, and this driver should support all those versions. Also remove references to this version number throughout this PR
```suggestion
compatible: "cdns,i2c"
```

---

Sure, will update